frontend/workflows/k8s/src/scale-resources.tsx

```diff
@@ -0,0 +1,289 @@
+  containerName: string;
+}): IClutch.k8s.v1.Deployment.DeploymentSpec.PodTemplateSpec.PodSpec.IContainer {
+  return args.deploymentSpec.template.spec.containers.find(
+    container => container.name === args.containerName
+  );
+}
+
+const Confirm: React.FC<ConfirmChild> = () => {
+  const deployment = useDataLayout("deploymentData").displayValue() as IClutch.k8s.v1.Deployment;
+  const updateData = useDataLayout("updateData");
+  const currentDeploymentData = useDataLayout(
+    "currentDeploymentData"
+  ).displayValue() as IClutch.k8s.v1.Deployment;
+
+  const updateRows: any[] = [];
+
+  let updatedContainer = false;
+  deployment.deploymentSpec.template.spec.containers.forEach(container => {
+    Object.keys(container.resources).forEach(resourceRequirement => {
+      Object.keys(container.resources[resourceRequirement]).forEach(resourceName => {
+        const newValue = container.resources[resourceRequirement][resourceName];
+        const oldValue = findContainer({
+          deploymentSpec: currentDeploymentData.deploymentSpec,
+          containerName: container.name,
+        }).resources[resourceRequirement][resourceName];
+        if (newValue !== oldValue) {
+          if (!updatedContainer) {
+            updateRows.push({ name: "Container Name", value: container.name });
+            updatedContainer = true;
+          }
+          updateRows.push({
+            name: `Old ${formatResourceString(resourceName, resourceRequirement)}`,
+            value: oldValue,
+          });
+          updateRows.push({
+            name: `New ${formatResourceString(resourceName, resourceRequirement)}`,
+            value: newValue,
+          });
+        }
+      });
+    });
+  });
+
+  return (
+    <WizardStep error={updateData.error} isLoading={updateData.isLoading}>
+      <Confirmation action="Update" />
+      <MetadataTable
+        data={[
+          { name: "Name", value: deployment.name },
+          { name: "Namespace", value: deployment.namespace },
+          { name: "Cluster", value: deployment.cluster },
+          ...updateRows,
+        ]}
+      />
+    </WizardStep>
+  );
+};
+
+const ScaleResources: React.FC<WorkflowProps> = ({ heading, resolverType }) => {
+  const dataLayout = {
+    inputData: {},
+    deploymentData: {},
+    currentDeploymentData: {},
+    updateData: {
+      deps: ["deploymentData", "inputData", "currentDeploymentData"],
+      hydrator: (
+        deploymentData: {
+          cluster: string;
+          containerName: string;
+          deploymentSpec: IClutch.k8s.v1.Deployment.DeploymentSpec;
+          name: string;
+          namespace: string;
+        },
+        inputData: { clientset: string },
+        currentDeploymentData: IClutch.k8s.v1.Deployment
+      ) => {
+        const clientset = inputData.clientset ?? "undefined";
+        const limits: { [key: string]: string } = {
+          cpu: findContainer({ ...deploymentData }).resources.limits.cpu,
```

Comment:
We should probably break this out into a variable instead of calling findContainer 4 times.

ie
```
const container = findContainer({ ...deploymentData });
```

```suggestion
          cpu: container.resources.limits.cpu,
```

Reply:
Updated.